### PR TITLE
HBW-79 Replace Select2 with react-select in widget

### PR DESCRIPTION
**Pull request by TimAle**
Waiting for #171 now.

### CODE REVIEW

File: hbw/app/javascript/packs/hbw/components/form/select.js.jsx [hunk 1]
```diff
@@ -49,120 +57,155 @@ modulejs.define('HBWFormSelect',
...
+          cursor:      'pointer',
+
+          '&:hover': {
+            borderColor: 'inherit',
+            borderWidth: 2
+          }
+        }),
+        option: (base, state) => ({
+          ...base,
+          color: (state.isFocused || state.isSelected)  ? 'white' : '#2C3E50',
```
Comment (sugarfree1) [inline hbw/app/javascript/packs/hbw/components/form/select.js.jsx]: Alignment

File: hbw/app/javascript/packs/hbw/components/form/select.js.jsx [hunk 2]
```diff
@@ -49,120 +57,155 @@ modulejs.define('HBWFormSelect',
...
+      const { inputValue } = options;
 
-      return select = e.find('select').select2(jQuery.extend({}, {
-        width:       '100%',
-        allowClear:  this.props.params.nullable,
-        theme:       'bootstrap',
-        placeholder: this.props.params.placeholder,
-        language:    this.getLanguage()
-      }, ajaxOptions)).on('change', () => this.setValue());
+      if (!inputValue || inputValue.length < 2) {
```
Comment (sugarfree1) [inline hbw/app/javascript/packs/hbw/components/form/select.js.jsx]: Lets reverse it.
```
if (inputValue && inputValue.length >= 2)
```

**Comment by TimAle** [comment]
UPD
Something new: not only `HBWFormSelect` uses Select2, there is the `HBWFormUser`.

File: hbw/app/javascript/packs/hbw/components/form/user.js.jsx [hunk 3]
```diff
@@ -1,50 +1,13 @@
...
-        },
-        formatSelection (node) {
-          return node.id;
-        },
-        cache: true
-      },
-      minimumInputLength: 1
+      mode:        'lookup',
+      url:         '/users/lookup',
+      choices:     [],
```
[the remark this answers is not in the document]
Comment (TimAle) [reply hbw/app/javascript/packs/hbw/components/form/user.js.jsx]: Are you sure? It's single-line value aligned with the longest key `placeholder`.

[the remark this answers is not in the document]
Comment (sugarfree1) [reply hbw/app/javascript/packs/hbw/components/form/user.js.jsx]: Missed it, sorry. It's ok of course.

**Pull request merged**